tests/core/test_core_device_containers.cpp

```diff
@@ -0,0 +1,39 @@
+/** VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "vecmem/containers/const_device_vector_data.hpp"
+#include "vecmem/containers/device_vector_data.hpp"
+
+// GoogleTest include(s).
+#include <gtest/gtest.h>
+
+// System include(s).
+#include <type_traits>
+
+/// Test case for the custom device container types
+class core_device_container_test : public testing::Test {
+
+}; // class core_device_container_test
+
+// Test that the "simple" data types are trivially constructible.
+TEST_F( core_device_container_test, trivial_construct ) {
```

Comment:
Can't we do these tests at compile time? Not that it really matters...

Reply:
:thinking: Are you thinking of adding some `static_assert(...)`-s? Sure, we could do that as well...

Reply:
It's a minor point, but catching these errors at compile time will surely make debugging a lot easier. We all know how... Unhelpful the C++ compiler can be when you make errors in your type-level programming. These tests might very well not even compile under certain error conditions.